beauty-consulting-app/app/Pages/Messaging/ChatPage.js

```diff
@@ -1,17 +1,215 @@
+import {
+	View,
+	Text,
+	StyleSheet,
+	FlatList,
+	TextInput,
+	Button,
+	TouchableOpacity,
+	KeyboardAvoidingView,
+	Platform,
+	ScrollView,
+} from "react-native";
 import SignupBackground from "../../assets/components/SignupBackground";
+import api from "utils/axios";
+import handleHTTPError from "utils/errorHandling";
+import { UserContext } from "../../contexts/userContext";
+import MessageBubble from "../../assets/components/MessageBubble";
 
 function ChatPage({ route }) {
-	const { username } = route.params;
+	const { username, stylistUsername, clientUsername } = route.params;
+	const userContext = useContext(UserContext);
+	const [messageHistory, setMessageHistory] = useState();
+	const [newMessage, setNewMessage] = useState("");
+	var ws = useRef(null);
+	const [isConnected, setIsConnected] = useState(false);
+
+	//username is the name of whomever is being chatted with
+	//stylistUsername and clientUsername are the literal
+	//usernames
+	//e.g. username = Stylist 7
+	//     clientUsername = stylist7
+	const fetchConversation = async () => {
+		try {
+			const response = await api.get("/messages/history", {
+				params: {
+					clientUsername: clientUsername,
+					stylistUsername: stylistUsername,
+				},
+			});
+			setMessageHistory(response.data);
+		} catch (error) {
+			handleHTTPError(error, setErrorMessage);
+		}
+	};
+
+	useFocusEffect(
+		React.useCallback(() => {
+			fetchConversation();
+		}, [])
+	);
+
+	useFocusEffect(
+		React.useCallback(() => {
+			//This is a bit of a hack to get the URL from
+			//EXPO_PUBLIC_API_URL
+			//Need to revisit
+			const apiUrl = process.env.EXPO_PUBLIC_API_URL;
+			const websocketUrl = apiUrl.replace(/^http:\/\//, "");
+			ws.current = new WebSocket(`ws://${websocketUrl}`);
+
+			ws.current.onopen = () => {
+				setIsConnected(true);
+				ws.current.send(
+					JSON.stringify({
+						event: "joinRoom",
+						username: clientUsername,
+					})
+				);
+				ws.current.send(
+					JSON.stringify({
+						event: "joinRoom",
+						username: stylistUsername,
+					})
+				);
+			};
+			ws.current.onmessage = (event) => {
+				const data = JSON.parse(event.data);
+				if (data.event === "messageReceived") {
+					setMessageHistory((prevMessages) => [
+						...prevMessages,
+						data.message,
+					]);
+				}
+			};
+			ws.current.onclose = () => {
+				setIsConnected(false);
+			};
+			return () => {
+				if (ws.current) {
+					ws.current.close();
+				}
+			};
+		}, [])
+	);
+
+	const sendMessage = () => {
+		if (
+			newMessage.trim() === "" ||
+			!ws.current ||
+			ws.current.readyState !== WebSocket.OPEN
+		) {
+			return;
+		}
+
+		const message = {
+			event: "sendMessage",
+			clientUsername: clientUsername,
+			stylistUsername: stylistUsername,
+			sender: userContext.username,
+			content: newMessage.trim(),
+			createdAt: new Date().toISOString(),
+		};
+
+		ws.current.send(JSON.stringify(message));
+		setMessageHistory((prevMessages) => [...prevMessages, message]);
```

Comment:
getting error when i try to send, from stack trace it seems to be here? something funky with message and prevMessages